### ISSUE TITLE
SSO doesn't work when accessing wp-admin on subsite prior to the front-end

### ISSUE THREAD

**Issue by jjeaton** [humanmade/Mercator#73]
Steps to reproduce:

1. Create a network with 2 subsites and (subsite1.example.dev and subsite2.example.dev) and map domain aliases (subsite1.dev and subsite2.dev).
2. Login to the first subsite (subsite1.dev).
3. Navigate to subsite2.dev/wp-admin/
4. Get redirected back to (subsite1.dev) without getting logged into subsite2.dev.

If you hit (subsite2.dev) prior to trying to access wp-admin, you are then logged in.

Is this intended behavior? Shouldn't hitting wp-admin also trigger the SSO flow to get you logged into the second site?

Site admins are irritated that whenever they use the "My Sites" menu to try to go to a different site's dashboard, they are redirected, until they hit Visit Site first.

**Comment by robrecord** [comment]
I resolved this by uninstalling Mercator, updating all references to the site URL in the database, and relying on WordPress' built-in domain redirection.

--
Rob Record
Web Designer/Developer
07535 271502 (tel:07535%20271502) | robrecord.com (http://robrecord.com)

**Comment by roborourke** [comment]
@jjeaton I'm just about to get to SSO as part of an overhaul of the Mercator ecosystem. Working out the best way to create tests for it. I was able to reproduce your problem and you're right it shouldn't behave in that way as it doesn't really make sense from a user perspective.

**Comment by roborourke** [comment]
@jjeaton so I'm clear are you using multinetwork too or just standard multisite?

**Comment by jjeaton** [comment]
@roborourke standard multisite

**Comment by jjeaton** [comment]
@roborourke any progress here? Will this issue get moved to the SSO repo when they split?

**Comment by roborourke** [comment]
Yeah all updates here will be ported. I'm going to fix #68 first then come
back to this one so no progress yet beyond replicating the issue I'm afraid.
